extensions/hc-dev-theme-extension/assets/hcBopis.js

```diff
@@ -0,0 +1,1036 @@
+    });
+    const data = await response.json();
+    lat = data?.response?.docs[0]?.latitude;
+    lon = data?.response?.docs[0]?.longitude;
+  } catch (error) {
+    console.error("Error fetching latitude and longitude:", error);
+  }
+  return { lat, lon };
+}
+
+async function searchStoresByLocation(lat, lon) {
+  console.log(`Searching stores by location - Latitude: ${lat}, Longitude: ${lon}`);
+  resetStoreList();
+  if (!lat || !lon) {
+    return;
+  }
+  const storeListContainer = STORE_LIST;
+  if (!storeListContainer) return;
+  storeListContainer.dataset.point = `${lat},${lon}`;
+  storeListContainer.dataset.viewIndex = 0;
+  await generateStoreListHTML(storeListContainer);
+  await initializePagination(storeListContainer);
+}
+
+async function searchStoresByZip(zipcode) {
+  const { lat, lon } = await getLatLon(zipcode);
+  await searchStoresByLocation(lat, lon);
+}
+
+function addToCart(currentVariantId, quantity = 1, properties) {
+  if (!currentVariantId) {
+    alert('No variant selected!');
+    return;
+  }
+  fetch('/cart/add.js', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      'X-Requested-With': 'XMLHttpRequest'
+    },
+    body: JSON.stringify({
+      items: [
+        {
+          id: currentVariantId,
+          quantity: quantity,
+          properties: properties
+        }
+      ]
+    })
+  })
+  .then(response => response.json())
+  .then(data => {
+    // TODO: Update cart UI
+    console.log('Added to cart:', data);
+    return fetch('/cart.js');
+  })
+  .then(cart => {
+    window.location.href = '/cart';
+  })
+  .catch(error => {
+    console.log("Error adding to cart:", error);
+  });
+}
+
+async function checkPickupInventory(payload) {
+  const response = await fetch('https://dev-maarg.hotwax.io/rest/s1/ofbiz-oms-usl/checkBopisInventory', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(payload)
+  });
+
+  const data = await response.json();
+  return data;
+}
+
+async function filterStoresByInventoryAvailability(stores, selectedVariantId) {
+  let checkBopisInventoryResult = [];
+  let storeCodes = [];
+  let storesWithInventory = [];
+  stores.forEach(
+    store => {
+      storeCodes.push(store.storeCode);
+    }
+  );
+  const payload = {};
+
+  payload.facilityIds = storeCodes;
+  payload.internalNames = [selectedVariantId];
+  payload.productStoreId = 'STORE';
+  payload.inventoryGroupId = 'SHOPIFY_1';
+
+  const response = await checkPickupInventory(payload);
+
+  if (response && response.resultList) {
+    checkBopisInventoryResult = response.resultList;
+    checkBopisInventoryResult.forEach(inv => {
+      if (inv.computedAtp > 0) {
+        storesWithInventory.push(inv.facilityId);
+      }
+    });
+  }
+  return storesWithInventory;
+}
+
+async function generateStoreListHTML(container) {
+  if (!container) return;
+  const enablePickup = Boolean(container.dataset.showPickupHere);
```

Comment:
Do not use Boolean for converting the values as this will return false only for falsy values, for 'false' this will return true.